doc/developer-guides/hld/hld-splitlock.rst

```diff
@@ -0,0 +1,122 @@
+********************************
+The Intel-64 and IA32 multiple-processor systems support locked atomic
+operations on locations in system memory. For example, The LOCK instruction
+prefix be prepended to the following instructions: ADD, ADC, AND, BTC, BTR, BTS,
+CMPXCHG, CMPXCH8B, CMPXCHG16B, DEC, INC, NEG, NOT, OR, SBB, SUB, XOR, XADD,
+XCHG. and these instructions must with memory destination operand forms.
+Reading or writing a byte in system memory are always guaranteed to be handled
+atomically, otherwise, these locked atomic operations can impact system in two
+ways:
+
+- The destination operand located in same cache line. Cache coherency protocols
+  that ensure that atomic operations can be carried out on cached data
+  structures with cache lock.
+
+- The destination operand located in two cache line. This atomic operation is
+  called a Split-locked Access. For this situation, LOCK# bus signal is asserted
+  to lock the system bus, to ensure they are atomic. See `Intel 64 and IA-32 Architectures Software Developer's Manual(SDM), Volume 3, (Section 8.1.2 Bus Locking) <https://software.intel.com/en-us/download/intel-64-and-ia-32-architectures-sdm-combined-volumes-3a-3b-3c-and-3d-system-programming-guide>`_.
+
+Split-locked Access may lead to long latency, to ordinary memory operations on
+other CPUs, which is hard to investigate.
+
+Split-locked Access Detection
+*****************************
+From the Microarchitecture Tremont, a new CPU capability is introduced for
+detection of Split-locked Access. When this feature is enabled, an alignment
+check exception(#AC) with error code 0 is raised for instructions which are
+doing Split-locked Access. #AC is a fault, the instruction is not executed,
+there is a chance to decide what to do with this instruction in the #AC handler:
+
+- Allow the instruction to run with LOCK# bus signal, may have impact to other
+  CPUs.
+- Disable LOCK# assertion for split locked access. That is not right because
+  instruction becomes non-atomic, and Intel plans to remove this CPU feature
+  from select products starting from a specific year. See `SDM, Volume 1, (Section 2.4 PROPOSED REMOVAL FROM UPCOMING PRODUCTS) <https://software.intel.com/en-us/download/intel-64-and-ia-32-architectures-software-developers-manual-volume-1-basic-architecture>`_.
+- Terminate the software at this instruction.
+
+Feature Enumeration and Control
+*******************************
+#AC for Split-locked Access feature is enumarated and controled via CPUID and
+MSR registers.
+
+- CPUID.(EAX=0x7, ECX=0):EDX[30], the 30th bit of output value in EDX indicates
+  if the platform has IA32_CORE_CAPABILITIES MSR.
+
+- The 5th bit of IA32_CORE_CAPABILITIES MSR(0xcf), enumerates whether the CPU
+  support #AC for Split-locked Access(And has TEST_CTRL MSR).
+
+- The 29th bit of TEST_CTL MSR(0x33) control enable/disable #AC for Split-locked
+  Access.
+
+Note that P6 family CPUs also has TEST_CTRL MSR, not enumerated in this way and
```

Comment:
The P6 family is very old now and ACRN does not run on it, it makes me wonder if this note is really relevant in this context?

Reply:
Thanks, will remove this part